### PR TITLE
Lexer: Omit Keywords and Variables Prefixes

### LINKED ISSUE #9: Lexer: Variable token should omit `$`
```go
item{typ: itemVariable, val: "$variable"} // current
item{typ: itemVariable, val: "variable"}  // expected
```

### PR DESCRIPTION
**Pull request by VladislavsPerkanuks**
closes #9 

~**waiting for #19 as conflicts are expected**~

**Also**:
- Omit `.` prefixes for keywords
- Refactor `lexName` to reflect changes + readability improvements?

### CODE REVIEW

File: lex.go [hunk 1]
```diff
@@ -370,32 +370,33 @@ func lexExpr(l *lexer) stateFn {
...
+		l.backup() // backup to the first rune
+	}
 
-		switch {
-		default:
-			l.backup()
+	var (
+		s string // item value
+		r rune   // current rune
+	)
```
Comment (janishorsts) [inline lex.go:389]:
```suggestion
	var s string // item value
```

File: lex.go [hunk 2]
```diff
@@ -370,32 +370,33 @@ func lexExpr(l *lexer) stateFn {
...
-		case len(s) == 0 && isNameStart(r):
-			s = string(r)
-		case len(s) == 0 && r == '$':
-			s = string(r)
-			typ = itemVariable
-		case len(s) == 0 && r == '.':
-			s = string(r)
-			typ = itemKeyword
-		}
+	for r = l.next(); isName(r); r = l.next() {
```
[the remark this answers is not in the document]
Comment (VladislavsPerkanuks) [reply lex.go:391]: cannot, r is needed outside the loop

File: lex.go (hunk 1 above)
Comment (VladislavsPerkanuks) [reply lex.go:389]: cannot do that, r is needed outside the loop

File: lex.go [hunk 3]
```diff
@@ -370,32 +370,33 @@ func lexExpr(l *lexer) stateFn {
...
+		s += string(r)
 	}
+
+	if r != eof { // backup is not necessary if we reached eof
```
Comment (janishorsts) [inline lex.go:395]: Isn't `eof` broken MF2 syntax?

Name is either a variable name or an unquoted literal.

Comment (VladislavsPerkanuks) [reply lex.go:395]: it is, but it has same behaviour as it was before

File: lex.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (janishorsts) [reply lex.go:391]: what about this?

```suggestion
	for r := l.next(); isName(r); r = l.next() {
	  if r == eof {
	    // unexpected eof
	  }
	}
	
	l.backup()
	
	// ...
```

[the remark this answers is not in the document]
Comment (janishorsts) [reply lex.go:391]: or what you have but error on eof

```go
if r == eof {
 // error on unexpected eof
}

l.backup()
```

[the remark this answers is not in the document]
Comment (VladislavsPerkanuks) [reply lex.go:391]: this could work, previous one I think will not, as eof will fail isName().

File: lex.go (hunk 3 above)
Comment (janishorsts) [reply lex.go:395]: Let's do it. Let's return the unexpected eof error.

**Review by janishorsts — APPROVED** [review]

**Pull request merged**